lib/kubeclient/common.rb

```diff
@@ -285,27 +292,35 @@ def self.underscore_entity(entity_name)
         .downcase
     end
 
-    def create_rest_client(path = nil)
-      path ||= @api_endpoint.path
+    def create_http_client(url = nil)
+      url = "#{@api_endpoint}/#{@api_version}" if url.nil?
       options = {
-        ssl_ca_file: @ssl_options[:ca_file],
-        ssl_cert_store: @ssl_options[:cert_store],
-        verify_ssl: @ssl_options[:verify_ssl],
-        ssl_client_cert: @ssl_options[:client_cert],
-        ssl_client_key: @ssl_options[:client_key],
         proxy: @http_proxy_uri,
-        max_redirects: @http_max_redirects,
-        user: @auth_options[:username],
-        password: @auth_options[:password],
-        open_timeout: @timeouts[:open],
-        read_timeout: @timeouts[:read]
+        request: {
+          open_timeout: @timeouts[:open],
+          read_timeout: @timeouts[:read]
+        },
+        ssl: {
+          ca_file: @ssl_options[:ca_file],
+          cert_store: @ssl_options[:cert_store],
+          client_cert: @ssl_options[:client_cert],
+          client_key: @ssl_options[:client_key],
+          verify: @ssl_options[:verify_ssl]
+        }
       }
-      RestClient::Resource.new(@api_endpoint.merge(path).to_s, options)
+
+      Faraday.new(url, options) do |connection|
+        if @auth_options[:username] && @auth_options[:password]
+          connection.basic_auth(@auth_options[:username], @auth_options[:password])
+        end
+        connection.use(FaradayMiddleware::FollowRedirects, limit: @http_max_redirects)
```

Comment:
is following redirects even a supported usecase / does restclient do it ?

Reply:
... could be a opt-in that then does a dynamic require to reduce dependencies

Reply:
Yes, `rest_client` does automatically handle redirects by default: https://github.com/rest-client/rest-client#redirection.

Is support for redirects needed for Kubernetes API server? I don't know. I wouldn't bet on not needing it though...

Reply:
faraday_middleware is 16.5 KB and brings no other deps.